### ISSUE TITLE
Requesting diagnostics on AKS/GKE does not work

### ISSUE THREAD

**Issue by michalpristas** [elastic/elastic-agent#2702]
When trying to request diagnostics from an agent running on k8s deployment on AKS it is not successful.

Root cause unknown at this point, it can be memory/cpu limits, timeout/proxy setting on AKS..

**Comment by Danouchka** [comment]
Same for GKE

**Comment by Danouchka** [comment]
cc @valerioarvizzigno @hemantmalik @eric-lowry

**Comment by cmacknz** [comment]
The issue description here is very vague. I don't know why the diagnostics functionality would be at all tied to the Kubernetes environment it is deployed into. 

What I think is much more likely is that collecting diagnostics triggers a spike in the memory usage of the agent that could cause it to be OOMKilled.

On an affected agent can we run `kubectl describe pod` and look for the following?

```
       Last State:   Terminated
       Reason:       OOMKilled
       Exit Code:    137
```

**Comment by Danouchka** [comment]
Indeed, I had to set the memory limit in the manifest to 1000Mi so that elastic-diagnostics can be extracted remotely from Kibana. 1GB RAM, that's so much

**Comment by cmacknz** [comment]
Thanks for confirming, agreed that having to increase the memory limit from 700Mi to 1000Mi just to allow requesting diagnostics is not ideal. There might be something we can do to improve this, perhaps we are holding too much of the diagnostics content in memory before writing it to disk.

**Comment by Danouchka** [comment]
On my GKE overloaded cluster, that would be nearly impossible to increase to that level, that would require spreading the applications on 2 or 3 other nodes to leave room for elastic-agent memory consumption
